### PR TITLE
SubscriberData can now throw errors on init

### PR DESCRIPTION
**Pull request by taquitos**
Add more custom error types.

### CODE REVIEW

File: Purchases/Public/CustomerInfo.swift [hunk 1]
```diff
@@ -158,17 +158,37 @@ import Foundation
...
+
+            let extraContext: String?
+            switch subscriberDataError {
+            case .originalAppUserIdMissing:
+                let originalAppUserId = subscriberObject["original_app_user_id"] as? String ?? "missing"
+                extraContext = "original_app_user_id is: \(originalAppUserId)"
+            case .firstSeenMissing, .firstSeenFormat:
+                let firstSeen = subscriberObject["first_seen"] as? String ?? "missing"
+                extraContext = "first_seen is: \(firstSeen)"
+            }
```
Comment (aboedo) [inline Purchases/Public/CustomerInfo.swift:189]: we could extract this into a method so we don't have to disable the swiftlint warning

File: Purchases/Public/CustomerInfo.swift [hunk 2]
```diff
@@ -221,7 +241,28 @@ import Foundation
         return parsePurchaseDates(transactionsByProductId: allPurchases)
     }()
 
-    private struct SubscriberData {
+    fileprivate struct SubscriberData {
+
+        // swiftlint:disable nesting
```
Comment (aboedo) [inline Purchases/Public/CustomerInfo.swift:246]: same note regarding `:next`

**Review by aboedo — APPROVED** [review]

File: Purchases/Public/CustomerInfo.swift [hunk 3]
```diff
@@ -158,17 +158,37 @@ import Foundation
                       transactionsFactory: TransactionsFactory())
     }
 
+    // swiftlint:disable function_body_length
```
[the remark this answers is not in the document]
Comment (taquitos) [reply Purchases/Public/CustomerInfo.swift:161]: TIL, that's awesome.

File: Purchases/Public/CustomerInfo.swift (hunk 1 above)
Comment (taquitos) [reply Purchases/Public/CustomerInfo.swift:189]: Good idea! Done! ✅

File: Purchases/Public/CustomerInfo.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (taquitos) [reply Purchases/Public/CustomerInfo.swift:161]: ✅

File: Purchases/Public/CustomerInfo.swift (hunk 2 above)
Comment (taquitos) [reply Purchases/Public/CustomerInfo.swift:246]: ✅

**Pull request merged**